source/slang/slang-check-modifier.cpp

```diff
@@ -918,6 +918,40 @@ namespace Slang
                 // See SemanticsDeclHeaderVisitor::checkExtensionExternVarAttribute.
             }
         }
+
+        if (auto packOffsetModifier = as<HLSLPackOffsetSemantic>(m))
+        {
+            if (!packOffsetModifier->registerName.getContent().startsWith("c"))
+            {
+                getSink()->diagnose(packOffsetModifier, Diagnostics::unknownRegisterClass, packOffsetModifier->registerName);
+                return m;
+            }
+            auto uniformOffset = stringToInt(packOffsetModifier->registerName.getContent().tail(1)) * 16;
+            if (packOffsetModifier->componentMask.getContentLength())
+            {
+                switch (packOffsetModifier->componentMask.getContent()[0])
+                {
+                case 'x':
+                    uniformOffset += 0;
+                    break;
+                case 'y':
+                    uniformOffset += 4;
+                    break;
+                case 'z':
+                    uniformOffset += 8;
+                    break;
+                case 'w':
+                    uniformOffset += 12;
+                    break;
```

Comment:
Do these depend on the component type being 4 bytes?

Reply:
I think so, it is specialized to xyzw according to the HLSL doc.